src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,94 @@
-export const Pagination = () => {};
+import { FC, useEffect, useState } from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number
+  perPage: number
+  currentPage: number
+  onPageChange: (page: number) => void
+};
+
+export const Pagination: FC<Props> = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const pages = getNumbers(1, Math.ceil(total / perPage));
+
+  const [prev, setPrev] = useState(currentPage - 1);
+  const [next, setNext] = useState(currentPage + 1);
+
+  useEffect(() => {
+    setNext(currentPage + 1);
+    setPrev(currentPage - 1);
+  }, [currentPage]);
+
+  const isLastPage = currentPage === pages.length;
+  const isFirstPage = currentPage === 1;
+
+  return (
+    <ul className="pagination">
+      <li
+        className={cn('page-item', {
+          disabled: isFirstPage,
+        })}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={isFirstPage}
+          onClick={() => {
+            if (!isFirstPage) {
+              onPageChange(prev);
+            }
+          }}
+        >
+          «
+        </a>
+      </li>
```

Comment:
prev and next pages are 95% the same - make a tiny function that will render the proper button
like this:
```
const renderStepButton = (direction: 'next' | 'prev') => {...}
```
and use this short function instead two bulky similar constructs
`{renderStepButton('next')}` / `{renderStepButton('prev')}`

Reply:
That sounds pretty cool. Thank you Roman